rosbridge_library/src/rosbridge_library/internal/message_conversion.py

```diff
@@ -112,41 +109,46 @@
     ("char[]", re.compile(r"char\[[^\]]*\]")),
 )
 
+module_configured = False
 binary_encoder = None
 binary_encoder_type = "default"
 bson_only_mode = False
 
 
-# TODO(@jubeira): configure module with a node handle.
-# The original code doesn't seem to actually use these parameters.
-def configure(node_handle: Node | None = None) -> None:
-    global binary_encoder, binary_encoder_type, bson_only_mode
+def configure(parameters: dict[str, Any] | None = None) -> None:
+    """
+    Configure the message_conversion module.
 
-    if node_handle is not None:
-        binary_encoder_type = (
-            node_handle.get_parameter_or("binary_encoder", Parameter("", value="default"))
-            .get_parameter_value()
-            .string_value
-        )
-        bson_only_mode = (
-            node_handle.get_parameter_or("bson_only_mode", Parameter("", value=False))
-            .get_parameter_value()
-            .bool_value
-        )
+    ;param parameters: A dictionary of parameters to configure the module.
+    :type parameters: dict[str, Any] | None
+    :raises ValueError: If an unknown encoder type is specified.
+    """
+    global binary_encoder, binary_encoder_type, bson_only_mode, module_configured
+
+    if module_configured:
+        return
+
+    if parameters is not None:
+        if "binary_encoder" in parameters:
+            binary_encoder_type = parameters["binary_encoder"]
```

Comment:
Rather than globals (yuck, though they were here before), I would do something like

```python
binary_encoder_type = parameters.get("binary_encoder", None)
```

(Of course the `, None` is not needed above, but putting there for other parameters that have different defaults

Reply:
Although I see that those globals are used by the other free functions, so are they necessary? Hmmn, what if `configure()` simply returns the values it needs?

Reply:
`binary_encoder_type` global is not used by any function so it can be removed

Reply:
I removed `binary_encoder_type`, one less global. `binary_encoder` and `bson_only_mode` are used by the functions though.

Reply:
I had an idea for a refactor which would add a `MessageConverter` class that would have the `extract_values` and `populate_instance` methods and would take the protocol parameters in the constructor. The `Protocol` class would create an instance of the converter and use it instead of calling the free functions. This would allow to configure the converter with different parameters per Protocol instance (allowing for example using BSON and JSON at the same time). 

When I started refactoring it, it turned out it would require changes and many places and I decided to not do this to not overcomplicate this PR.